controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -443,7 +452,33 @@ func (r *KubeadmControlPlaneReconciler) computeDesiredMachine(kcp *controlplanev
 	if existingMachine != nil {
 		desiredMachine.Spec.InfrastructureRef = existingMachine.Spec.InfrastructureRef
 		desiredMachine.Spec.Bootstrap.ConfigRef = existingMachine.Spec.Bootstrap.ConfigRef
+		desiredMachine.Spec.ReadinessGates = existingMachine.Spec.ReadinessGates
 	}
+	ensureMandatoryReadinessGates(desiredMachine)
 
 	return desiredMachine, nil
 }
+
+func ensureMandatoryReadinessGates(m *clusterv1.Machine) {
+	if m.Spec.ReadinessGates == nil {
+		m.Spec.ReadinessGates = mandatoryMachineReadinessGates
+		return
+	}
+
+	additionalConditions := []clusterv1.MachineReadinessGate{}
+
+	for _, want := range mandatoryMachineReadinessGates {
+		found := false
+		for _, got := range m.Spec.ReadinessGates {
+			if got.ConditionType == want.ConditionType {
+				found = true
+				break
+			}
+		}
+		if !found {
+			additionalConditions = append(additionalConditions, want)
```

Comment:
```suggestion
			m.Spec.ReadinessGates = append(m.Spec.ReadinessGates, want)
```
nit: just to simplify a bit